src/parser/tokenizer/index.ts

```diff
@@ -218,6 +218,12 @@ export function lookaheadTypeAndKeyword(): TypeAndKeyword {
   return new TypeAndKeyword(type, contextualKeyword);
 }
 
+// Returns true if the next character is a valid start for an identifier
+export function nextIsIdentifier(): boolean {
```

Comment:
Opted for this instead of accessing `input` in the expression traverser, which is essentially what babel does. Should be simple to switch around if you prefer something different.

Reply:
Never mind, ended up syncing more from babel-parser, which moves over to accessing `input` in the traverser